### PR TITLE
feat: add util for weighted routes

### PR DESCRIPTION
**Pull request by stahntobias**

### CODE REVIEW

**Comment by theSuess** [comment]
Please run `tk fmt .` in the project root to pass the failing CI test

File: util/route.libsonnet [hunk 1]
```diff
@@ -17,6 +17,31 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
...
+      d.arg('path', d.T.string, '/'),
+    ]),
+    weightedRouteFor(name, host, port, services, path='/')::
+      route.new(name, host, path)
+      + route.spec.to.withKind('Service')
+      + route.spec.to.withName(services[0].name)
+      + route.spec.to.withWeight(services[0].weight)
+      + route.spec.port.withTargetPort(port)
+      + route.spec.withAlternateBackends(
+        std.prune(
```
Comment (theSuess) [inline util/route.libsonnet:38]: I think using slice syntax here might be more readable (had to look up the documentation for `prune` to understand what exactly is happening here)

For example `services[1:std.length(services):1]`

File: util/route.libsonnet [hunk 2]
```diff
@@ -17,6 +17,31 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
...
+          std.mapWithIndex(
+            function(i, v) if i == 0 then null else v { kind: 'Service' }, services
```
Comment (theSuess) [inline util/route.libsonnet:40]: According to the docs, the `kind` field defaults to `Service`. Have you tested leaving this empty?

**Review by theSuess — CHANGES_REQUESTED** [review]

File: util/route.libsonnet (hunk 2 above)
Comment (stahntobias) [reply util/route.libsonnet:40]: Yes. when leaving empty one gets
`error: error validating "STDIN": error validating data: ValidationError(Route.spec.alternateBackends[0]): missing required field "kind" in com.github.openshift.api.route.v1.RouteTargetReference; if you choose to ignore these errors,
 turn validation off with --validate=false` during tk apply

Comment (theSuess) [reply util/route.libsonnet:40]: Okay, then let's keep this

**Review by theSuess — APPROVED** [review]

**Pull request merged**